level3.cc

```diff
@@ -0,0 +1,180 @@
+
+    // checking is a move puts my king in check
+    for (int i = 0; i < arraySize; ++i) {
+        std::map<Box, int> potentialMoves = (*((*(pieces[i])).getLegalMoves()));
+        for (auto &move: potentialMoves) { 
+
+            // create temporary pieces that will be moved around and then deleted
+            Piece *tempPiece;
+            if (((*pieces[i])).getName() == "p" || (*(pieces[i])).getName() == "P") {
+                tempPiece = new Pawn {(*(pieces[i])).getName(), subject->getBoard(), (*(pieces[i])).checkWhitePlayer(), (*(pieces[i])).getX(), (*(pieces[i])).getY()};
+            } else if ((*(pieces[i])).getName() == "r" || (*(pieces[i])).getName() == "R") {
+                tempPiece = new Rook {(*(pieces[i])).getName(), subject->getBoard(), (*(pieces[i])).checkWhitePlayer(), (*(pieces[i])).getX(), (*(pieces[i])).getY()};  
+            } else if ((*(pieces[i])).getName() == "q" || (*(pieces[i])).getName() == "Q") {
+                tempPiece = new Queen {(*(pieces[i])).getName(), subject->getBoard(), (*(pieces[i])).checkWhitePlayer(), (*(pieces[i])).getX(), (*(pieces[i])).getY()};
+            } else if ((*(pieces[i])).getName() == "k" || (*(pieces[i])).getName() == "K") {
+                tempPiece = new King {(*(pieces[i])).getName(), subject->getBoard(), (*(pieces[i])).checkWhitePlayer(), (*(pieces[i])).getX(), (*(pieces[i])).getY()};             
+            } else if ((*(pieces[i])).getName() == "n" || (*(pieces[i])).getName() == "N") {
+                tempPiece = new Knight {(*(pieces[i])).getName(), subject->getBoard(), (*(pieces[i])).checkWhitePlayer(), (*(pieces[i])).getX(), (*(pieces[i])).getY()};               
+            } else {
+                tempPiece = new Bishop {(*(pieces[i])).getName(), subject->getBoard(), (*(pieces[i])).checkWhitePlayer(), (*(pieces[i])).getX(), (*(pieces[i])).getY()};          
+            }
+
+            // if the current move is a capture we need to not loose the piece it will capture and bring it back after checking for king check
+            if ((*(subject->getBoard()))[(move.first).getX()][(move.first).getY()]) {
+                Piece *temp = (*(subject->getBoard()))[(move.first).getX()][(move.first).getY()];
+                (*(subject->getBoard()))[(move.first).getX()][(move.first).getX()] = tempPiece;
+                (*(subject->getBoard()))[(*(pieces[i])).getX()][(*(pieces[i])).getY()] = nullptr;
+                subject->checkingForKingCheck();
+
+                // if my king is not in check after potential move is made -> add to newMap
+                if (((*(pieces[i])).checkWhitePlayer() && !(subject->isWhiteKingChecked())) || 
+                    (!((*(pieces[i])).checkWhitePlayer()) && !(subject->isBlackKingChecked()))) {
+
+                        if (find(toAvoid.begin(), toAvoid.end(), move.first) == toAvoid.end()) {
+                            // move makes current piece captured
+                            filteredMap[&(*(pieces[i]))] = move.first;
```

Comment:
`&(*(pieces[i]))` is dereferencing and then taking the address again, which is the same as `pieces[i]`

Reply:
ohhhh shooot ill fix this rn